src/pages/AddCar.jsx

```diff
@@ -1,8 +1,236 @@
-const AddCar = () => (
-  <div>
-    <h1>Add Car form</h1>
-    <p>The add car form goes here</p>
-  </div>
-);
-
-export default AddCar;
+import React, { useEffect, useState } from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+import axios from 'axios';
+import { fetchAppData } from 'app/redux/AppDataSlice';
+
+export default function AddCar() {
+  const engineTypes = useSelector((store) => store.appData.appData.engine_type);
+  const dispatch = useDispatch();
+  useEffect(() => {
+    dispatch(fetchAppData);
+  }, [dispatch]);
+  const [formData, setFormData] = useState({
+    name: '',
+    description: '',
+    image_url: '',
+    engine_type_id: engineTypes[0].id,
+    horsepower: '',
+    torque: '',
+    fuel_economy: '',
+    range: '',
+    seating_capacity: '',
+    cargo_space: '',
+    infotainment_system: '',
+    safety_rating: '',
+    tech_features: '',
+    special_features: '',
+  });
+
+  const handleChange = (e) => {
+    const { name, value } = e.target;
+    setFormData({ ...formData, [name]: value });
+  };
+
+  const handleSubmit = async (e) => {
+    e.preventDefault();
+    try {
+      const formDataWithImage = new FormData();
+      formDataWithImage.append('name', formData.name);
+      formDataWithImage.append('description', formData.description);
+      formDataWithImage.append('image', formData.image);
+      formDataWithImage.append(
+        'car_detail[engine_type_id]',
+        formData.engine_type_id,
+      );
+      formDataWithImage.append('car_detail[horsepower]', formData.horsepower);
+      formDataWithImage.append('car_detail[torque]', formData.torque);
+      formDataWithImage.append(
+        'car_detail[fuel_economy]',
+        formData.fuel_economy,
+      );
+      formDataWithImage.append(
+        'car_detail[seating_capacity]',
+        formData.seating_capacity,
+      );
+      formDataWithImage.append('car_detail[cargo_space]', formData.cargo_space);
+      formDataWithImage.append(
+        'car_detail[infotainment_system]',
+        formData.infotainment_system,
+      );
+      formDataWithImage.append(
+        'car_detail[safety_rating]',
+        formData.safety_rating,
+      );
+      formDataWithImage.append(
+        'car_detail[tech_features]',
+        formData.tech_features,
+      );
+      formDataWithImage.append(
+        'car_detail[special_features]',
+        formData.special_features,
+      );
+
+      await axios.post('http://localhost:4000/api/v1/cars', formDataWithImage);
```

Comment:
When doing any type of request that requires authorization, you'll need to pass the access token as the header. You can get the accesstoken from 'state.auth.token'.

here is an example of how I did it when creating reservations:

`const response = await fetch('http://localhost:4000/api/v1/reservations', {
        method: 'POST',
        headers: {
          'Content-Type': 'application/json',
          Authorization: `Bearer ${accessToken}`,
        },
        body: JSON.stringify({
          reservation: {
            user_id: userId,
            car_id: carId,
            city_id: city,
            date,
          },
        }),
      });`